utils/common/src/main/java/brooklyn/util/crypto/AuthorizedKeysParser.java

```diff
@@ -0,0 +1,132 @@
+
+public class AuthorizedKeysParser {
+
+    public static PublicKey decodePublicKey(String keyLine) {
+        try {
+            ByteArrayInputStream stream = null;
+
+            // look for the Base64 encoded part of the line to decode
+            // both ssh-rsa and ssh-dss begin with "AAAA" due to the length bytes
+            for (String part : keyLine.split(" ")) {
+                if (part.startsWith("AAAA")) {
+                    stream = new ByteArrayInputStream(BaseEncoding.base64().decode(part));
+                    break;
+                }
+            }
+            if (stream == null)
+                throw new IllegalArgumentException("Encoded public key should include phrase beginning AAAA.");
+
+            String type = readType(stream);
+            if (type.equals("ssh-rsa")) {
+                BigInteger e = readBigInt(stream);
+                BigInteger m = readBigInt(stream);
+                RSAPublicKeySpec spec = new RSAPublicKeySpec(m, e);
+                return KeyFactory.getInstance("RSA").generatePublic(spec);
+            } else if (type.equals("ssh-dss")) {
+                BigInteger p = readBigInt(stream);
+                BigInteger q = readBigInt(stream);
+                BigInteger g = readBigInt(stream);
+                BigInteger y = readBigInt(stream);
+                DSAPublicKeySpec spec = new DSAPublicKeySpec(y, p, q, g);
+                return KeyFactory.getInstance("DSA").generatePublic(spec);
+            } else {
+                throw new IllegalArgumentException("Unknown public key type " + type);
+            }
+            
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            throw new IllegalArgumentException("Error parsing authorized_keys/SSH2 format public key: "+e);
+        }
+    }
+
+    private static int readInt(InputStream stream) throws IOException {
+        return ((stream.read() & 0xFF) << 24) | ((stream.read() & 0xFF) << 16)
+            | ((stream.read() & 0xFF) << 8) | (stream.read() & 0xFF);
+    }
+    
+    private static byte[] readBytesWithLength(InputStream stream) throws IOException {
+        int len = readInt(stream);
+        byte[] result = new byte[len];
+        stream.read(result);
+        return result;
+    }
+    
+    private static void writeInt(OutputStream stream, int v) throws IOException {
+        for (int shift = 24; shift >= 0; shift -= 8)
+            stream.write((v >>> shift) & 0xFF);
+    }
+    private static void writeBytesWithLength(OutputStream stream, byte[] buf) throws IOException {
+        writeInt(stream, buf.length);
+        stream.write(buf);
+    }
+    
+    private static String readType(InputStream stream) throws IOException { return new String(readBytesWithLength(stream)); }
+    private static BigInteger readBigInt(InputStream stream) throws IOException { return new BigInteger(readBytesWithLength(stream)); }
+
+    public static String encodePublicKey(PublicKey key) {
```

Comment:
Surprised there's not some utility method in one of our dependencies that already does this, but I haven't looked at all.